File: 11-container-with-most-water/container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    let end = height.length-1;
    let start = 0;
    let maxw = 0;
    let w = 0;
    while(start!=end){
        length = end - start;
        w = Math.min(height[start], height[end]) * length;
        maxw = Math.max(w,maxw)
        if (height[start]<height[end])
        start++;
        else
        end--;
    }
    return maxw;
};

File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        start = 0
        end = len(height)-1
        mw = 0
        while start!=end:
            length = end - start
            w = (min(height[start],height[end])) * length
            mw = max(w,mw)
            if height[start]<height[end]:
                start+=1
            else:
                end-=1
        return mw
        